Projects/NUCLEO-H743ZI/Applications/LwIP/LwIP_HTTP_Server_Netconn_RTOS/MDK-ARM/STM32H743ZI-Nucleo/STM32H743ZI-Nucleo.sct

```diff
@@ -11,16 +11,15 @@ LR_IROM1 0x08000000 0x00200000  {    ; load region size_region
   RW_IRAM2 0x24000000 0x00080000  {  ; RW data
    .ANY (+RW +ZI)
   }
-  RW_DMARxDscrTab 0x30040000 0x60 {
+  RW_DMARxDscrTab 0x30000000 0x60 {
   *(.RxDecripSection)
   }
-  RW_DMATxDscrTab 0x30040060 0x140 {
+  RW_DMATxDscrTab 0x30000200 0x140 {
   *(.TxDecripSection)
   }
-  RW_Rx_Buffb 0x30040200 0x1800 {
-  *(.RxArraySection)
+  memory_RX_POOL_base 0x30000400 0x12676 {
```

Comment:
In the lwipopts.h you set the lwip heap pointer to address 0x3000 4000 (Line 54) and here you use 0x3000 0400.
Is this correct?

Reply:
Hi @KnoerleMan,

Thank you for your review and for this remark. I will transfer it to our development teams and will get back to you with their answer.

With regards,

Reply:
Hi again,

seems like i maybe understood why the address of the "LwIP_HeapSection" in the scatter file is different from the LWIP_RAM_HEAP_POINTER define in the lwipopts.h file.
The heap section at address 0x30000400 is for the memory pool of the RxBuffers. The required size is about 12K (12 buffers * size of 1000) plus the additional overhead of the RxBuff_t structure.

The heap for the lwip stack (start address = LWIP_RAM_HEAP_POINTER, size = MEM_SIZE) is therefore placed at location 0x30004000 which means the memory pool of the RxBuffers can take up 15360 bytes, which should be enough.
This heap has nothing to do with the scatter file section "LwIP_HeapSection". The lwip heap seems to be used for some Tx operations.

Could someone try to clarify if this is correct?

Reply:
Hi @KnoerleMan,

You guessed it right! According to our development teams:
* the Rx pool (named `LwIP_HeapSection`) is used for **packets reception** whereas,
* the `LWIP_RAM_HEAP_POINTER` is used for the **LwIP heap**.

The name `LwIP_HeapSection` is clearly confusing and the proposal is to change it into something like `Rx_PoolSection` in both `ethernetif.c` and scatter files. An internal ticket will be logged to have this fixed.

Thank you for having reported this point.

With regards,

Reply:
ST Internal Reference: 121723